mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -600,6 +607,21 @@ private void updateHeartbeatTimeout(TaskExecutorID taskExecutorID) {
             heartbeatTimeout);
     }
 
+    private void onCacheJobArtifactsOnTaskExecutorRequest(CacheJobArtifactsOnTaskExecutorRequest request) {
+        log.info("[fdc-91] onCacheJobArtifactsRequest: {}", request);
+        TaskExecutorState state = this.executorStateManager.get(request.getTaskExecutorID());
+        if (state != null) {
+            TaskExecutorGateway gateway = state.getGateway().join();
+            try {
+                List<URI> artifacts = mantisJobStore.getJobArtifactsToCache(request.getClusterID());
```

Comment:
shall the actor store/cache this instead of making a call to store on every TE registration? Also it can use this state to determine whether it needs to trigger an cacheRequest message in the registration call directly?

Reply:
I don't see that many benefits in caching this very tiny info in memory which I'm sure can be retrieved from the datastore in milliseconds.

Reply:
In current logic during the upgrade this read will happen to every TE including the ones don't have any cache request. Just seems to be many unnecessary cycles concurrently in IO executor though.

Reply:
added cache.